Goal: make the -f option force the removal of the user's group (even if it
      is the primary group of another user)

Status wrt upstream: Fixed in 4.0.14

Index: shadow-4.0.13/src/userdel.c
===================================================================
--- shadow-4.0.13.orig/src/userdel.c	2005-10-12 00:47:01.000000000 +0200
+++ shadow-4.0.13/src/userdel.c	2005-10-12 01:00:39.000000000 +0200
@@ -172,6 +172,9 @@
 	if (grp && getdef_bool ("USERGROUPS_ENAB")
 	    && (grp->gr_mem[0] == NULL)) {
 
+		pwd = NULL;
+		if (!fflg)
+		{
 		/*
 		 * Scan the passwd file to check if this group is still
 		 * used as a primary group.
@@ -189,6 +192,7 @@
 			}
 		}
 		endpwent ();
+		}
 
 		if (pwd == NULL) {
 			/*
